sidecar/internal/util/util.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2022 The Kubernetes-CSI-Addons Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+)
+
+// ValidateControllerEndpoint validate endpoint to make sure it contains
+// valid ip address and port number.
+func ValidateControllerEndpoint(endpoint string) error {
+	separator := ":"
```

Comment:
Actually, this will fail on IPv6 addresses. Those should definitely be supported!

Reply:
> Actually, this will fail on IPv6 addresses. Those should definitely be supported!

modified to handle IPv6 too, and added some tests, PTAL @nixpanic @Madhu-1